funcframework/framework.go

```diff
@@ -37,20 +38,27 @@ const (
 	errorStatus          = "error"
 )
 
+var (
+	handler = http.DefaultServeMux
+)
+
 // RegisterHTTPFunction registers fn as an HTTP function.
-func RegisterHTTPFunction(path string, fn interface{}) {
-	fnHTTP, ok := fn.(func(http.ResponseWriter, *http.Request))
-	if !ok {
-		panic("expected function to have signature func(http.ResponseWriter, *http.Request)")
-	}
-	registerHTTPFunction(path, fnHTTP, http.DefaultServeMux)
+func RegisterHTTPFunction(path string, fn func(http.ResponseWriter, *http.Request)) error {
```

Comment:
Confirming: this is a breaking change. Is that OK?

Reply:
I'm going to need to release a v2 anyways for the new signature type, so might as well get all them breaking changes in now.